Tests/CatalogItems/Get-RsSubscription.Tests.ps1

```diff
@@ -0,0 +1,335 @@
+    
+    $proxy = New-RsWebServiceProxy
+    $namespace = $proxy.GetType().NameSpace
+
+    $ExtensionSettingsDataType = "$namespace.ExtensionSettings"
+    $ParameterValueOrFieldReference = "$namespace.ParameterValueOrFieldReference[]"
+    $ParameterValueDataType = "$namespace.ParameterValue"
+
+    #Set ExtensionSettings
+    $ExtensionSettings = New-Object $ExtensionSettingsDataType
+                    
+    $ExtensionSettings.Extension = "Report Server FileShare"
+
+    #Set ParameterValues
+    $ParameterValues = New-Object $ParameterValueOrFieldReference -ArgumentList 7
+
+    $to = New-Object $ParameterValueDataType
+    $to.Name = "PATH";
+    $to.Value = "\\unc\path"; 
+    $ParameterValues[0] = $to;
+
+    $replyTo = New-Object $ParameterValueDataType
+    $replyTo.Name = "FILENAME";
+    $replyTo.Value ="Report";
+    $ParameterValues[1] = $replyTo;
+
+    $includeReport = New-Object $ParameterValueDataType
+    $includeReport.Name = "FILEEXTN";
+    $includeReport.Value = "True";
+    $ParameterValues[2] = $includeReport;
+
+    $renderFormat = New-Object $ParameterValueDataType
+    $renderFormat.Name = "USERNAME";
+    $renderFormat.Value = "user";
+    $ParameterValues[3] = $renderFormat;
+
+    $priority = New-Object $ParameterValueDataType
+    $priority.Name = "RENDER_FORMAT";
+    $priority.Value = "PDF";
+    $ParameterValues[4] = $priority;
+
+    $subject = New-Object $ParameterValueDataType
+    $subject.Name = "WRITEMODE";
+    $subject.Value = "Overwrite";
+    $ParameterValues[5] = $subject;
+
+    $comment = New-Object $ParameterValueDataType
+    $comment.Name = "DEFAULTCREDENTIALS";
+    $comment.Value = "False";
+    $ParameterValues[6] = $comment;
+
+    $ExtensionSettings.ParameterValues = $ParameterValues
+
+    $subscription = [pscustomobject]@{
+        DeliverySettings      = $ExtensionSettings
+        Description           = "Shared on \\unc\path"
+        EventType             = "TimedSubscription"
+        IsDataDriven          = $false
+	    MatchData             = $matchData.OuterXml
+    }
+    
+    return $subscription
+}
+
+Describe "Get-RsSubscription" {
+        Context "Get-RsSubscription without parameters"{
+                $folderName = 'SutGetRsItemReference_MinParameters' + [guid]::NewGuid()
+                New-RsFolder -Path / -FolderName $folderName
+                $folderPath = '/' + $folderName
+                $localResourcesPath =   (Get-Item -Path ".\").FullName  + '\Tests\CatalogItems\testResources\emptyReport.rdl'
+                Write-RsCatalogItem -Path $localResourcesPath -RsFolder $folderPath
+                $report = (Get-RsFolderContent -RsFolder $folderPath )| Where-Object TypeName -eq 'Report'
+                
+                $subscription = Get-NewFileShareSubscription
+
+                $localResourcesPath =   (Get-Item -Path ".\").FullName  + '\Tests\CatalogItems\testResources\UnDataset.rsd'
```

Comment:
as the setup for the report is common between the functions and is lengthy , I recommend to move it to a function like you did with the Get-NewFileShareSubscription